src/sapling/noteencryption.cpp

```diff
@@ -299,9 +300,9 @@ NoteEncryption<MLEN>::NoteEncryption(uint256 hSig) : nonce(0), hSig(hSig) {
     // All of this code assumes crypto_scalarmult_BYTES is 32
     // There's no reason that will _ever_ change, but for
     // completeness purposes, let's check anyway.
-    BOOST_STATIC_ASSERT(32 == crypto_scalarmult_BYTES);
-    BOOST_STATIC_ASSERT(32 == crypto_scalarmult_SCALARBYTES);
-    BOOST_STATIC_ASSERT(NOTEENCRYPTION_AUTH_BYTES == crypto_aead_chacha20poly1305_ABYTES);
+    static_assert(32 == 32U, "32 == crypto_scalarmult_SCALARBYTES");
+    static_assert(32 == 32U, "32 == crypto_scalarmult_SCALARBYTES");
+    static_assert(16 == 16U, "NOTEENCRYPTION_AUTH_BYTES == crypto_aead_chacha20poly1305_ABYTES");
```

Comment:
Why this?
The purpose of the assertion is to ensure that `crypto_scalarmult_SCALARBYTES` is 32 (same for the others), you here removed the assertion and now is checking if an int 32 is equal to a unsigned integer 32 (which is always true).

Should be:
```
static_assert(32 == crypto_scalarmult_BYTES, "invalid crypto_scalarmult_BYTES");
static_assert(32 == crypto_scalarmult_SCALARBYTES, "invalid crypto_scalarmult_SCALARBYTES");
static_assert(NOTEENCRYPTION_AUTH_BYTES == crypto_aead_chacha20poly1305_ABYTES, "noteencryp_auth_bytes !=  crypto_aead_chacha20poly1305_ABYTES");



```

Reply:
updated